Added instructions on booting Pi5 from NVMe SSD (#2225)
Screenly/Anthias: docs/raspberry-pi5-ssd-install-instructions.md

# How to install on a Raspberry Pi 5 with PCI-e SSD

> ## Appropriate Hardware

Reviewer: ```suggestion
## Hardware
```

> The following guide has been tested using a Pi5/8GB and a GeeekPi P33 PoE+PCI-e HAT. (See: https://pipci.jeffgeerling.com/hats/geeekpi-p33-m2-nvme-poe-hat.html )

Reviewer: Please remove the space before the closing parentheses.

```suggestion
The following guide has been tested using a Pi5/8GB and a GeeekPi P33 PoE+PCI-e HAT. (See: https://pipci.jeffgeerling.com/hats/geeekpi-p33-m2-nvme-poe-hat.html)
```

Reviewer: Alternatively, you could embed the link in "GeeekPi P33 PoE+PCI-e HAT".

```suggestion
The following guide has been tested using a Raspberry Pi 5 with 8GB RAM and a [GeeekPi P33 PoE+PCI-e HAT](https://pipci.jeffgeerling.com/hats/geeekpi-p33-m2-nvme-poe-hat.html).
```

Other HAT's should work just fine as long as they are supported by the Pi.

> We used a 256GB M.2 NVMe SSD as it came in the 2242 form factor, and was small and cheap (at time of writing, these were approx £15+vat from our supplier).

Reviewer: ```suggestion
We used a 256GB M.2 NVMe SSD as it came in the 2242 (22mm x 44mm) form factor.
```

> You may also need a SD Card initially to configure your Pi, once the first steps are done you won't require this.

Reviewer: ```suggestion
You may also need a microSD card to initially configure your Pi. Once the first steps are done you won't require this.
```

> ## Install procedure

Reviewer: ```suggestion
## Installation
```

> Using the Raspberry Pi Imager and appropriate USB adapters, write the 64-bit version of Raspberry Pi OS to both a SD card and the SSD.

Reviewer: ```suggestion
Using the Raspberry Pi Imager and appropriate USB adapters, write the 64-bit version of Raspberry Pi OS Lite (Bookworm) to both the microSD card and the SSD.
```

> There are other ways (Such as a network boot if enabled on the Pi, Booting from SD and using the SD Card copier utility to copy the OS to the SSD, or Booting from SD and using the RPI Imager to write a fresh copy to the SSD, or using rpiboot mode to present the SSD as mass storage to a PC.

Reviewer: ```suggestion
There alternative ways to install:
- Network boot (if enabled on the Pi)
- Booting from the microSD card and using the SD Card copier utility to copy the OS to the SSD
- Booting from the microSD and using the Raspberry Pi Imager to write a fresh copy to the SSD
- Using `rpiboot` mode to display the SSD as a mass storage device on a PC.
```

Do whatever is easiest for you!

> ### Boot from SD

Reviewer: ### Boot from the microSD Card

> Depending on the bootloader version of your Pi, you may need to boot from SD first and set the bootloader to boot from PCIe.

Reviewer: ```suggestion
Depending on the bootloader version of your Pi, you may need to boot from microSD first and set the bootloader to boot from PCIe.
```

> - Once booted, run the RPI EEPROM configurator: `sudo rpi-eeprom-config -edit`

Reviewer: ```suggestion
- Once booted, run the RPI EEPROM configurator: `sudo rpi-eeprom-config -edit`. This will open up the [Nano](https://www.nano-editor.org/) text editor.
```

- Change the boot order to: `BOOT_ORDER=0xf416`
- Add the line: `PCIE_PROBE=1`

> - Type: Ctrl-O to save the file

Reviewer: ```suggestion
- Type **Ctrl-O** to save the file.
```

> - Type: Ctrl-X to exit the editor

Reviewer: ```suggestion
- Type **Ctrl-X** to exit the editor.
```

> - Remove the SD card, and power cycle the Pi.

Reviewer: ```suggestion
- Remove the microSD card, and power cycle the Pi.
```

### Boot from SSD

Once your Pi is booting from the SSD, you have a couple of housekeeping tasks to perform;

> - Update the OS using `sudo apt update` and `sudo apt full-upgrade -y`

Reviewer: ```suggestion
- Update the OS using `sudo apt update -y` and `sudo apt full-upgrade -y`
```

- Check the bootloader is at the latest version using `sudo rpi-eeprom-update`

> - If there is a eeprom update, then use `sudo raspi-config` to update it

Reviewer: ```suggestion
- If there is an EEPROM update, then use `sudo raspi-config` to update it
```

> - Go to `6 Advanced Opitions` => `A5 Bootloader Version` => `E1 Latest`, then select `Yes`

Reviewer: ```suggestion
- Go to `6 Advanced Opitions` &rarr; `A5 Bootloader Version` &rarr; `E1 Latest`, then select `Yes`
```

`&rarr;` translates to a right arrow symbol.

> - Reboot the Pi

Reviewer: ```suggestion
- Reboot the Pi.
```

- Finally, you'll need to run the Anthias installer: `bash <(curl -sL https://install-anthias.srly.io)`
- Follow the prompts to install and reboot the Pi
- Once the install has completed, don't forget to change the password for your Pi uinsg `passwd`

> ### Post install issues

Reviewer: ```suggestion
### Post Install Issues
```

> If after it completes and you reboot you still get a black screen, simply press `control+alt+f1` to get into the console and then run: `./screenly/bin/upgrade_containers.sh` this should re-run the container creation step and have the system up and running properly.

Reviewer: ```suggestion
> [!NOTE]
> If you still get a black screen after the installation completes and after a reboot, simply press `Ctrl-Alt-F1` to get into the console and then run `./screenly/bin/upgrade_containers.sh`. this should re-run the container creation step and have the system up and running properly.
```
